### PR TITLE
Introduce inventory app

### PR DESCRIPTION
**Pull request by alex-voodoo**

### CODE REVIEW

**Review by anarute — COMMENTED** [review]
I also forgot to comment the other day that we really need the status field. It's super important to know the status of an asset, if it's with someone, if it's broken in the office, available to be used, etc.

I like the approach of using the type `CharField` with choices pre-defined by constants to restrict the options the user can select for this field:

```python
ALLOCATED = "allocated"
AVAILABLE = "available"
BROKEN = "broken"
DISCARDED = "discarded"
RETURNED_TO_CUSTOMER = "returned"
STORED = "stored"

DEVICE_STATUSES = [
    (ALLOCATED, "Allocated"),
    (AVAILABLE, "Available"),
    (BROKEN, "Broken"),
    (DISCARDED = "discarded"),
    (RETURNED_TO_CUSTOMER, "Returned to Customer"),
    (STORED, "Stored"),
]

status = models.CharField(
    max_length=20,
    choices=DEVICE_STATUSES,
    default=ALLOCATED,
)
```

I normally store constants like this (simulating enums) in a global file because it tends to grow together with the application and also the same constant can be used in multiple places and avoid hardcoding strings.

File: inventory/models.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+class Device(models.Model):
+    """
+    A single item in the inventory of the company.
+
+    Properties of particular devices should be stored in a separate model associated with the DeviceType instance.
+    """
+    model = models.ForeignKey(DeviceModel, on_delete=models.PROTECT)
+
+    # The device may be assigned to someone.
+    owner = models.ForeignKey(Person, null=True, blank=True, on_delete=models.SET_NULL)
```
[the remark this answers is not in the document]
Comment (alex-voodoo) [reply inventory/models.py:43]: Renamed it to `assignee`.

**Comment by alex-voodoo** [comment]
Could the status and all the dates go to the device log?

**Comment by anarute** [comment]
> Could the status and all the dates go to the device log?

Not really, they are properties of the device item, it doesn't seem practical to have to run through the logs to get when an item was purchased or what's its current status. The log makes more sense to record when the device was updated and what kind of change, for instance when it changed status or when it changed assignee, but I see it being used only for logging and reporting purposes, not to store properties of the item.

**Comment by alex-voodoo** [comment]
Okay, that makes sense.

**Comment by alex-voodoo** [comment]
Thank you for the review!

**Pull request merged**